zk-token-sdk/src/range_proof/inner_product.rs

```diff
@@ -76,18 +84,20 @@ impl InnerProductProof {
         // If it's the first iteration, unroll the Hprime = H*y_inv scalar mults
         // into multiscalar muls, for performance.
         if n != 1 {
-            n /= 2;
+            n = n.checked_div(2).unwrap();
             let (a_L, a_R) = a.split_at_mut(n);
             let (b_L, b_R) = b.split_at_mut(n);
             let (G_L, G_R) = G.split_at_mut(n);
             let (H_L, H_R) = H.split_at_mut(n);
 
-            let c_L = util::inner_product(a_L, b_R);
-            let c_R = util::inner_product(a_R, b_L);
+            let c_L = util::inner_product(a_L, b_R)
+                .ok_or(RangeProofGenerationError::InnerProductLengthMismatch)?;
+            let c_R = util::inner_product(a_R, b_L)
+                .ok_or(RangeProofGenerationError::InnerProductLengthMismatch)?;
 
             let L = RistrettoPoint::multiscalar_mul(
                 a_L.iter()
-                    .zip(G_factors[n..2 * n].iter())
+                    .zip(G_factors[n..n.checked_mul(2).unwrap()].iter())
```

Comment:
Can you add comments explaining why all of the checked math unwraps are safe?

Reply:
Yep, done.